test/Models/CsvCatalogItemSpec.js

```diff
@@ -94,31 +111,318 @@ describe('CsvCatalogItem', function() {
+            expect(ip).toBeDefined();
+        }).otherwise(fail).then(done);
+    });
+
+    it('supports lat-long files with dates', function(done) {
+        csvItem.url = 'test/csv/lat_long_enum_moving_date.csv';
+        csvItem.load().then(function() {
+            var j = JulianDate.fromIso8601;
+            var source = csvItem._tableDataSource;
+            expect(source.dataset.getRowCount()).toEqual(13);
+            expect(csvItem._regionMapped).toBe(false);
+            expect(source.dataset.hasTimeData()).toBe(true);
+            expect(source.getDataPointList(j('2015-07-31')).length).toBe(0);
+            expect(source.getDataPointList(j('2015-08-01')).length).toBe(2);
+            expect(source.getDataPointList(j('2015-08-02')).length).toBe(3);
+            expect(source.getDataPointList(j('2015-08-06')).length).toBe(2);
+            expect(source.getDataPointList(j('2015-08-07')).length).toBe(0);
+        }).otherwise(fail).then(done);
+    });
+    it('supports lat-long files with dates and very long displayDuration', function(done) {
+        csvItem.url = 'test/csv/lat_long_enum_moving_date.csv';
+        csvItem.tableStyle = { displayDuration: 60 * 24 * 7 }; // 7 days
+        csvItem.load().then(function() {
+            var j = JulianDate.fromIso8601;
+            var source = csvItem._tableDataSource;
+            expect(source.dataset.getRowCount()).toEqual(13);
+            expect(csvItem._regionMapped).toBe(false);
+            expect(source.dataset.hasTimeData()).toBe(true);
+            expect(source.getDataPointList(j('2015-07-31')).length).toBe(0);
+            expect(source.getDataPointList(j('2015-08-01')).length).toBe(2);
+            expect(source.getDataPointList(j('2015-08-02')).length).toBe(5);
+            expect(source.getDataPointList(j('2015-08-06')).length).toBe(13);
+            expect(source.getDataPointList(j('2015-08-07')).length).toBe(13);
+        }).otherwise(fail).then(done);
+    });
+    it('supports lat-long files with dates sorted randomly', function(done) {
+        csvItem.url = 'test/csv/lat_lon_enum_moving_date_unsorted.csv';
+        csvItem.load().then(function() {
+            var j = JulianDate.fromIso8601;
+            var source = csvItem._tableDataSource;
+            expect(source.dataset.getRowCount()).toEqual(13);
+            expect(csvItem._regionMapped).toBe(false);
+            expect(source.dataset.hasTimeData()).toBe(true);
+            expect(source.getDataPointList(j('2015-07-31')).length).toBe(0);
+            expect(source.getDataPointList(j('2015-08-01')).length).toBe(2);
+            expect(source.getDataPointList(j('2015-08-02')).length).toBe(3);
+            expect(source.getDataPointList(j('2015-08-06')).length).toBe(2);
+            expect(source.getDataPointList(j('2015-08-07')).length).toBe(0);
+        }).otherwise(fail).then(done);
+    });
+    /*
+    Nope - I don't know how feature picking on lat-longs works.
```

Comment:
Yeah, it's tricky.  Lat-long files create actual vector points (attached to Cesium entities), which are then rendered and picked differently in the two clients:

* In Leaflet, `LeafletVisualizer` turns the point into a circle marker, which is an actual browser DOM element.  The browser detects a click on the circle marker DOM element and raises an event on `LeafletScene` that causes `featurePicked` in `Leaflet` to be invoked.
* In Cesium, our `Cesium.js` handles left clicks by calling `drillPick` on Cesium's `Scene`.  `drillPick` does one or more actual renders of the scene to an off-screen buffer in a fancy way so that the color of the rendered pixel indicates the ID of the vector feature that was picked.

Neither of these can be tested without the actual Leaflet / Cesium machinery up and running.

So, instead of testing picking, I recommend that you simply check that entities with the correct properties were created (one of my comments above has an example of checking the color of an entity's point) and trust that correctly-created entities will be rendered and will be pickable.